### PR TITLE
Add syscall rt_sigsuspend

### PR DESCRIPTION
**Pull request by StevenJiang1110**

### CODE REVIEW

File: services/libs/aster-std/src/syscall/rt_sigsuspend.rs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    // Set sigmask of current thread
+    let sigmask: SigMask = read_val_from_user(sigmask_addr)?;
+    let current_thread = current_thread!();
+    let poxis_thread = current_thread.as_posix_thread().unwrap();
+    *poxis_thread.sig_mask().lock() = sigmask;
+
+    // Pause until receiving any signal
+    let pauser = Pauser::new();
+    pauser.pause_until(|| None::<()>)?;
```
Comment (tatetian) [inline services/libs/aster-std/src/syscall/rt_sigsuspend.rs:31]: What if signals are already delivered to this thread (but not handled yet), before `pauser.pause_until` is called?

File: services/libs/aster-std/src/syscall/rt_sigsuspend.rs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    // This syscall should always return `Err(EINTR)`. This path should never be reached.
+    debug_assert!(false);
+    Ok(SyscallReturn::Return(0))
```
Comment (tatetian) [inline services/libs/aster-std/src/syscall/rt_sigsuspend.rs:35]: Replace the two lines with `unreachable!`

File: services/libs/aster-std/src/syscall/rt_sigsuspend.rs (hunk 1 above)
Comment (StevenJiang1110) [reply services/libs/aster-std/src/syscall/rt_sigsuspend.rs]: The `pause_until` will check whether there is any pending signal at first, so in this case, `pause_until` will return `EINTR`.

**Review by tatetian — APPROVED** [review]
LGTM. Thanks for the contribution!

**Comment by tatetian** [comment]
@StevenJiang1110 Need to resolve the conflict by rebasing to the latest main branch. Thanks

**Comment by StevenJiang1110** [comment]
This PR is rebased

**Pull request merged**